### PR TITLE
INFRA-8626 Alpha state abstraction for config updater

### PR DESCRIPTION
**Pull request by zunkree**

### CODE REVIEW

File: pkg/consul/watcher.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+type WatcherConfig struct {
+	RetryInterval  time.Duration
+	MaxBackoffTime time.Duration
+}
+
+type Callback func([]byte) error
+
+func NewWatcher(client *api.Client, config *WatcherConfig) *Watcher {
```
Comment (ebedarev) [inline pkg/consul/watcher.go]: Have you considered using https://github.com/hashicorp/consul/tree/master/sdk/testutil to test Watcher?

File: pkg/consul/client.go [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package consul
+
+import (
+	"github.com/hashicorp/consul/api"
+)
+
+func NewClient(address string) (*api.Client, error) {
+	config := api.DefaultConfig()
```
Comment (ebedarev) [inline pkg/consul/client.go]: Will there be a way to override default config options besides address or we won't need it?

Comment (zunkree) [reply pkg/consul/client.go]: for now nope, will be considered during further development

File: pkg/consul/watcher.go (hunk 1 above)
Comment (zunkree) [reply pkg/consul/watcher.go]: Not yet, but in future releases for sure.

**Pull request merged**